package.py

```diff
@@ -0,0 +1,911 @@
+            )
+            exit(1)
+
+        if not self.is_ci() and not os.environ.get('VIRTUAL_ENV'):
+            logger.error('The script should not run outside a virtual environment if not on CI')
+            exit(1)
+
+        self.check_repo()
+
+    def is_mac(self) -> bool:
+        return self.os == 'Darwin'
+
+    def is_linux(self) -> bool:
+        return self.os == 'Linux'
+
+    def is_windows(self) -> bool:
+        return self.os == 'Windows'
+
+    def is_mac_runner(self) -> bool:
+        return self.is_mac() and self.is_ci()
+
+    def is_universal2(self) -> bool:
+        return self.is_mac_runner() or os.environ.get('MACOS_BUILD_ARCH') == 'universal2'
+
+    def is_x86_64(self) -> bool:
+        return self.arch in ['x86_64', 'AMD64']
+
+    def os_tag(self) -> str:
+        if self.is_mac():
+            return 'macos'
+        if self.is_linux():
+            return 'linux'
+        if self.is_windows():
+            return 'windows.exe'
+        raise Exception(f'Invalid os {self.os}')
+
+    @staticmethod
+    def mac_electron_env_set() -> bool:
+        return os.environ.get('MACOS_ELECTRON_ARCH') is not None
+
+    def get_frontend_env(self) -> dict[str, str]:
+        if os.environ.get('GH_TOKEN'):
+            logger.info('GH_TOKEN WAS SET')
+        else:
+            logger.info('NO GH_TOKEN SET')
+        env = os.environ.copy()
+        if self.is_mac() and not self.mac_electron_env_set() and not self.is_ci():
+            if self.is_x86_64():
+                arch = 'x64'
+            else:
+                arch = 'arm64'
+
+            env.setdefault('MACOS_ELECTRON_ARCH', arch)
+        return env
+
+    @staticmethod
+    def is_ci() -> bool:
+        return os.environ.get('CI') is not None
+
+    @staticmethod
+    def update_path(directory: Path) -> None:
+        path = os.environ.get('PATH')
+        os.environ.setdefault('PATH', f'{path};{directory}')
+
+
+class Checksum:
+    @staticmethod
+    def generate(env: Environment, path: Path) -> Path:
+        checksum_filename = f'{path.name}.sha512'
+        cmd = None
+        if env.is_mac():
+            cmd = f'shasum -a 512 {path} > {checksum_filename}'
+        elif env.is_linux():
+            cmd = f'sha512sum {path} > {checksum_filename}'
+        elif env.is_windows():
+            cmd = f'powershell.exe -command "Get-FileHash {path} -Algorithm SHA512 | Select-Object Hash | foreach {{$_.Hash}} | Out-File -FilePath {checksum_filename}"'  # noqa E501
+        else:
+            logger.error('unsupported system')
+            exit(1)
+
+        ret_code = subprocess.call(cmd, shell=True)
+        if ret_code != 0:
+            logger.error(f'could not generate sha512 sum for {path}')
+            exit(1)
+        return path.parent.joinpath(checksum_filename)
+
+
+class Storage:
+    def __init__(self) -> None:
+        self.working_directory = Path(os.getcwd())
+        self.dist_directory = self.working_directory.joinpath('dist')
```

Comment:
In the rotki repo we are using the syntax with `/` instead of joinpath

```suggestion
        self.dist_directory = self.working_directory / 'dist'
```

Reply:
And does this work fine on windows?

Reply:
Yes, is a Pathlib operator https://docs.python.org/3/library/pathlib.html#operators